src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1346,8 +1346,16 @@ private ImmutableArray<TypeWithAnnotations> BindTypeArguments(SeparatedSyntaxLis
 
         private TypeWithAnnotations BindTypeArgument(TypeSyntax typeArgument, BindingDiagnosticBag diagnostics, ConsList<TypeSymbol> basesBeingResolved = null)
         {
-            // Unsafe types can never be type arguments, but there's a special error code for that.
-            var binder = this.WithAdditionalFlags(BinderFlags.SuppressUnsafeDiagnostics);
+            // BackCompat.  The compiler would previously suppress reporting errors for pointers in generic types.  This
+            // was intended so you would get a specific error in CheckBasicConstraints.CheckBasicConstraints for a type
+            // like (like `List<int*>`). i.e. you would get the error about an unsafe type not being a legal type argument,
+            // but not the error about not being in an unsafe context.  This had the unfortunate consequence though of 
+            // preventing the latter check for something like `List<int*[]>`.  Here, this is a legal generic type, but we 
+            // still want to report the error that you need to be in an unsafe context.  So, to maintain compat, we only
+            // do the suppression if you're on C# 11 and prior.  In later versions we do the correct check.
+            var binder = ((CSharpParseOptions)typeArgument.SyntaxTree.Options).LanguageVersion < MessageID.IDS_FeatureUsingTypeAlias.RequiredVersion()
```

Comment:
Seems this could be simplified:
```suggestion
            var binder = DeclaringCompilation.IsFeatureEnabled(MessageID.IDS_FeatureUsingTypeAlias)
```